sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -86,6 +86,24 @@ export interface SendOptions {
   abortSignal?: AbortSignalLike;
 }
 
+/**
+ * The set of options to configure the createBatch operation on the `Sender`.
+ */
+export interface BatchOptions {
+  /**
+   * @property
+   * A value that is hashed to produce a partition assignment.
+   * It guarantees that messages with the same partitionKey end up in the same partition.
+   * Specifying this will throw an error if the producer was created using a `paritionId`.
```

Comment:
We are currently throwing this error in the `send` operation. The comment here suggests that the error will be thrown when creating the batch.

Also, as a user I would like to see this error thrown when creating batch, instead of letting me create a batch with multiple events and throwing error at a later stage i.e send()

Thoughts?

Reply:
This need not block this PR. But if we do decide to go down the above route, please log an issue to track the same

Reply:
Agreed! 
Created an issue here: https://github.com/Azure/azure-sdk-for-js/issues/4302 to track this.